### PR TITLE
implement dbt MVP, both models and infrastructure

### PR DESCRIPTION
**Pull request by atvaccaro**
See [the relevant TDD](https://docs.google.com/document/d/17G89089wpzGGUoud41VMbId3XubUubcvsKbAMJ289fc/edit#) for the _why_ of this PR.

## Infra changes https://github.com/cal-itp/data-infra/pull/1265
1. Set up a dbt project and poetry project to support it
2. Build and deploy a docker image that contains the project
3. Create an Airflow pod operator to execute the dbt project on k8s
4. Deletes the Airflow tasks that create the migrated tables (see below).

## Modelling changes https://github.com/cal-itp/data-infra/pull/1310
1. Creates models to replace views.gtfs_rt_fact_files and views.gtfs_rt_fact_files_wide_hourly as part of the dbt MVP (initial models). These tables were selected because they are relatively functionally isolated and simple.
2. Changes the gtfs_rt schema to views -- in the interest of being able to migrate existing content, I think we can change this later

## Data pipeline schedule considerations
Currently this operator is scheduled to run at 12 UTC, so we can more easily debug at the beginning of each day. Eventually we will want this operator to run at least 2-3 times per day, so we can exercise the code regularly and prevent engineers from having to manually re-run tasks that may flake during an otherwise once-daily execution.

## Successful testing!
```
[2022-04-04 16:39:01,951] {pod_launcher.py:149} INFO - 16:39:02  Done. PASS=2 WARN=0 ERROR=0 SKIP=0 TOTAL=2
[2022-04-04 16:39:04,155] {pod_launcher.py:198} INFO - Event: dbt-run-and-upload-artifacts.8f4783abd2384b64b377ec1883666b64 had an event of type Succeeded
[2022-04-04 16:39:04,155] {pod_launcher.py:311} INFO - Event with job id dbt-run-and-upload-artifacts.8f4783abd2384b64b377ec1883666b64 Succeeded
[2022-04-04 16:39:04,260] {pod_launcher.py:198} INFO - Event: dbt-run-and-upload-artifacts.8f4783abd2384b64b377ec1883666b64 had an event of type Succeeded
[2022-04-04 16:39:04,260] {pod_launcher.py:311} INFO - Event with job id dbt-run-and-upload-artifacts.8f4783abd2384b64b377ec1883666b64 Succeeded
[2022-04-04 16:39:04,417] {taskinstance.py:1219} INFO - Marking task as SUCCESS. dag_id=transform_warehouse, task_id=dbt_run_and_upload_artifacts, execution_date=20220328T000000, start_date=20220329T154500, end_date=20220404T163904
```

### CODE REVIEW

**Review by lottspot — APPROVED** [review]
The kubernetes & ci aspects look good! :+1:

File: warehouse/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Welcome to your new dbt project!
```
Comment (lauriemerrell) [inline warehouse/README.md:1]: Should we delete this file, or at least make it relevant to our project? (Ex. at least reference `poetry run dbt run`)

File: warehouse/dbt_project.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+
+# Name your project! Project names should contain only lowercase characters
+# and underscores. A good package name should reflect your organization's
+# name or the intended use of these models
```
Comment (lauriemerrell) [inline warehouse/dbt_project.yml:4]:
```suggestion
```

Suggest we delete?

File: warehouse/dbt_project.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  - "target"
+  - "dbt_packages"
+
+
+# Configuring models
+# Full documentation: https://docs.getdbt.com/docs/configuring-models
+
+# In this example config, we tell dbt to build all models in the example/ directory
+# as tables. These settings can be overridden in the individual model files
+# using the `{{ config(...) }}` macro.
```
Comment (lauriemerrell) [inline warehouse/dbt_project.yml:33]:
```suggestion
```

Also suggest we delete

File: warehouse/macros/generate_schema_name.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{% macro generate_schema_name(custom_schema_name, node) -%}
```
Comment (lauriemerrell) [inline warehouse/macros/generate_schema_name.sql:1]: optional: Maybe some explanation (comment) would be useful here to explain what is going on

**Review by lauriemerrell — CHANGES_REQUESTED** [review]
Thanks Andrew-- a few documentation things that I think it would be good to tidy a bit before merge:

* I think this sentence from PR description can be deleted, right?
> It originally includes the skeleton of the dbt project, with two upcoming PRs to add dbt content and https://github.com/cal-itp/data-infra/pull/1265 separately.
* ~Would also add a note to description to say that this PR deletes the Airflow DAG tasks that generated these rt files views.~ edit sorry I see you already note this

File: .github/workflows/dbt-build.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+name: Build and push dbt image
+
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - 'warehouse/**'
```
Comment (themightychris) [inline .github/workflows/dbt-build.yml:8]: Always a good idea to include the workflow itself too:

```suggestion
      - '.github/workflows/dbt-build.yml'
      - 'warehouse/**'
```

**Review by themightychris — APPROVED** [review]
Docker and GitHub Actions bits look good, assuming my one suggestion gets merged

File: .github/workflows/dbt-build.yml (hunk 5 above)
Comment (atvaccaro) [reply .github/workflows/dbt-build.yml:9]: oh yeah thanks, I will do that on the gtfs-rt-parser one too

File: warehouse/macros/generate_schema_name.sql (hunk 4 above)
Comment (atvaccaro) [reply warehouse/macros/generate_schema_name.sql:1]: yes I will link to the dbt docs

File: warehouse/README.md (hunk 1 above)
Comment (atvaccaro) [reply warehouse/README.md:1]: Hm yeah let me add some actually useful content

**Review by lauriemerrell — APPROVED** [review]

**Pull request merged**